src/main/java/com/google/alpollo/SongInfoServlet.java

```diff
@@ -0,0 +1,45 @@
+package com.google.alpollo;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.alpollo.model.SongInfo;
+import com.google.gson.Gson;
+
+/** Servlet created to deal with songInfo related operations. */
+@WebServlet("/song-info")
+public class SongInfoServlet extends HttpServlet {
+  private static final String SONG_ID = "id";
+  private static final String SONG_INFO = "songInfo";
+  private final Gson gson = new Gson();
+
+  /**
+   * Making a GET request to our servlet with the desired song ID as a parameter
+   * will return that song from the database.
+   */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    int songId = gson.fromJson(request.getParameter(SONG_ID), int.class);
+    SongInfo songInfo = SongDataBase.getSongInfo(songId);
+
+    response.setContentType("application/json;");
+    try {
+      response.getWriter().println(gson.toJson(songInfo));
+    } catch (IOException e) {
+      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR,
+        e.getMessage());
+    }
+  }
+
+  /**
+   * Making a POST request to the server with the songInfo object as a parameter (object containing
+   * all the info related to our song) will send the song to the storage layer. 
+   */
+  @Override 
+  public void doPost(HttpServletRequest request, HttpServletResponse response) {
+    SongInfo songInfo = gson.fromJson(request.getParameter(SONG_INFO), SongInfo.class);
+    SongDataBase.saveSongInfo(songInfo);
```

Comment:
This method can throw an exception if something with objectify went wrong, so we should handle it and send the error in that case

Reply:
I had no idea. Do you mean `saveSongInfo()`?

Reply:
I don't know objectify, I just saw that, for example, `ofy()` can throw `IllegalStateExpression`. I think you or @sashapff should read the documentation and see what can exceptions can occur.

I don't think that we are able to catch specific types of exception since they depend on objectify, so maybe it's enough just to put this code to `try/catch` block and send the exception message if it occurred.

Reply:
It's reasonable. I will fix the same issue in the TopSongs Servlet as well.